lizmap/server/lizmap_accesscontrol.py

```diff
@@ -0,0 +1,284 @@
+        """ Are we authorise to modify the following geometry """
+        return super().allowToEdit(layer, feature)
+
+    def cacheKey(self) -> str:
+        """ The key used to cache documents """
+        defaultCacheKey = super().cacheKey()
+
+        # Get Lizmap user groups provided by the request
+        groups = self.getLizmapGroups()
+
+        # If groups is empty, no Lizmap user groups provided by the request
+        # The default cache key is returned
+        if len(groups) == 0:
+            return defaultCacheKey
+
+        # Get Lizmap config
+        cfg = self.getLizmapConfig()
+        if not cfg:
+            # Lizmap config is empty
+            QgsMessageLog.logMessage("Lizmap config is empty", "lizmap", Qgis.Warning)
+            # The default cache key is returned
+            return defaultCacheKey
+
+        # Check Lizmap config layers
+        if 'layers' not in cfg or not cfg['layers']:
+            # Lizmap config has no options
+            QgsMessageLog.logMessage("Lizmap config has no layers", "lizmap", Qgis.Warning)
+            # The default cache key is returned
+            return defaultCacheKey
+
+        # Check group_visibility in Lizmap config layers
+        cfg_layers = cfg['layers']
+        hasGroupVisibility = False
+        for l_name, cfg_layer in cfg_layers.items():
+            # check group_visibility in config
+            if 'group_visibility' not in cfg_layer or not cfg_layer['group_visibility']:
+                continue
+
+            # get group_visibility as list
+            group_visibility = cfg_layer['group_visibility'].split(',')
+            group_visibility = [g.strip() for g in group_visibility]
+
+            # the group_visibility was just an empty string
+            if len(group_visibility) == 1 and groups[0] == '':
+                continue
+
+            hasGroupVisibility = True
+            break
+
+        # group_visibility option is defined in Lizmap config layers
+        if hasGroupVisibility:
+            # The group provided in request is anonymous
+            if len(groups) == 1 and groups[0] == '':
+                return '@@'
+            # for other groups, removing duplicates and joining
+            return '@@'.join(list(set(groups)))
+
+        return defaultCacheKey
+
+    def getLizmapConfig(self) -> 'Dict':
+        """ Get Lizmap config """
+        # Get QGIS Project path
+        config_path = self.iface.configFilePath()
+        if not os.path.exists(config_path):
+            # QGIS Project path does not exist as a file
+            # No Lizmap config
+            return None
+
+        # Get Lizmap config path
+        config_path += '.cfg'
+        if not os.path.exists(config_path):
+            # Lizmap config path does not exist
+            QgsMessageLog.logMessage("Lizmap config does not exist", "lizmap", Qgis.Info)
+            # No Lizmap config
+            return None
+
+        # Get Lizmap config
+        cfg = None
```

Comment:
```suggestion
```

Reply:
Why this suggestion ?

Reply:
Because this variable is not used if you read the code. Either cfg is set L242 or you have an exception your code will return L246.

Reply:
And it's not only me :)
![Selection_002](https://user-images.githubusercontent.com/1609292/83166405-2b39a180-a10f-11ea-9d50-e6153a7d6bfe.png)

Reply:
I have update the code